### ISSUE TITLE
displayName JSX

### ISSUE THREAD

**Issue by Cellule** [yannickcr/eslint-plugin-react#75]
Would it be possible to add an option to displayName to consider the jsx transpiler. What I mean is with something like
`"displayName": [1, "jsx"]`

The following patterns are considered warnings:
```js
module.exports = React.createClass({
  render: function() {
    return <div>Hello {this.props.name}</div>;
  }
});
```

The following patterns are not considered warnings:
```js
var Hello = React.createClass({
  render: function() {
    return <div>Hello {this.props.name}</div>;
  }
});
```

**Comment by zertosh** [comment]
@Cellule FYI: Your first example would not get a `displayName` if you're transpiling with facebook's transform (react-tools/jstransform/reactify/etc). With babel, when using ES6 module syntax, it'll infer the `displayName` from the filename. http://babeljs.io/docs/usage/jsx/#additional-changes

**Comment by Cellule** [comment]
I didn't know that babel would add it from the file name, it's good to know. However, I am not using babel everywhere yet and I think it would still be useful to have a rule to force the developer to assign a name to its components even when using jsx. 
I suppose the rule could check for `export default` and not warn when no variable nor display name is specified

**Comment by yannickcr** [comment]
Currently the `display-name` rule enforce `displayName` to be set whenever you're using JSX or not.

What you suggesting is a more sloppy mode when the rule only enforce `displayName` if we directly export the component with `module.exports` ?

If yes, I think this is something feasible.

**Comment by Cellule** [comment]
yes, that's pretty much what I mean. The idea is to make sure you have a displayName even when transpiling jsx
